app/models/game.rb

```diff
@@ -29,29 +29,210 @@ def populate_board!
   end
 
   def in_check?(color)
-    # Query for remaining friendly pieces
-    player_pieces = pieces.where(game_id: id, color: color, captured?: false)
-
-    # Query for enemy king piece
-    enemy_king = pieces.where(game_id: id, type: 'King').where.not(color: color).last
-
-    # Store enemy_king coordinates for shorter reference
-    ekx = enemy_king.x_coordinate
-    eky = enemy_king.y_coordinate
-
-    player_pieces.each do |player_piece|
-      # Check if it would be valid for the current player_piece to move to the enemy king's position
-      if player_piece.valid_move? ekx, eky
-        # Check if the current player_piece wouldn't obstructed while attempting to move to the enemy king's position
-        unless player_piece.is_obstructed? ekx, eky
-          # If both criteria are met, at least once player_piece is successfully 'checking' the enemy king
+    set_remaining_piece_data color
+    
+    @remaining_player_pieces.each do |player_piece|
+    # Check if it would be valid for the current player_piece to move to the opponent king's position
+      if player_piece.valid_move? @okx, @oky
+      # Check if the current player_piece wouldn't obstructed while attempting to move to the opponent king's position
+        unless player_piece.is_obstructed? @okx, @oky
```

Comment:
You can reduce the length of this conditional by a couple lines if you used 'next if' instead of 'unless' like you did on line 71.